source/documentation/appendix-docs/botor.md

```diff
@@ -283,10 +323,75 @@ s3_path_to_full_df(
     sheet = 1)
 ```
 
+#### `s3_path_to_preview_df`
+
+```r
+s3_path_to_preview_df = function(s3_path, ...) {
+  s3_path <- stringr::str_replace(s3_path, "s3://", "")
+  split_path <- stringr::str_split(s3_path, "/")[[1]]
+  bucket <- split_path[1]
+  key <- stringr::str_c(split_path[-1], collapse="/")    
+  fext <- tolower(tools::file_ext(p$key))
+  if (!(fext %in% c("csv", "tsv"))) {
+    message(stringr::str_glue("Preview not supported for {fext} files"))
+    NULL
+  } else {
+    tryCatch(
+      {
+        client <- botor::botor()$client("s3")
+        obj <- client$get_object(Bucket = bucket, Key = key,
+                                 Range = "bytes=0-12000")
+        obj$Body$read()$decode() %>%
+          textConnection() %>%
+          read.csv() %>%
+          head(n = 5)
+      },
+      error = function(c) {
+        message("Could not read ", s3_path)
+        stop(c)
+      }
+    )
+  }
+}
+```
+
+##### Examples
+
+```r
+s3_path_to_preview_df('alpha-mybucket/my_data.csv')
+```
+
+#### `download_file_from_s3`
+
+```r
+download_file_from_s3 <- function(s3_path, local_path, overwrite = FALSE) {
+  # trim s3:// if included by the user
+  s3_path <- gsub('^s3://',"",s3_path)
+  # add s3:// back in where required
+  s3_path <- paste0('s3://', s3_path)
```

Comment:
```suggestion
  # trim s3:// if included by the user and add it back in where required
  s3_path <- paste0("s3://", gsub('^s3://', "", s3_path))
```

Reply:
Just to clean down the code slightly and make it slightly more obvious why this is being done

Reply:
b426b4d